components/Snackbar/src/private/MDCSnackbarOverlayView.m

```diff
@@ -211,6 +221,34 @@ - (void)setupContainerConstraints {
   [self addConstraint:self.bottomConstraint];
 }
 
+- (void)updateConstraints {
+  [super updateConstraints];
+
+  self.maximumHeightConstraint.constant = self.maximumHeight;
+
+  CGFloat sideMargin = [self sideMargin];
+  CGFloat leftMargin = sideMargin;
+  CGFloat rightMargin = sideMargin;
+
+  BOOL isRegularWidth = self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClassRegular;
+  BOOL isRegularHeight = self.traitCollection.verticalSizeClass == UIUserInterfaceSizeClassRegular;
```

Comment:
It’s not difficult to imagine that Apple might have notches in compact orientations in the future; can we remove these conditions?

Reply:
The reason this logic was added is because of the width of a Snackbar in devices that are of a regular size class doesn't go to full width and therefore does not collide with the safe area

Reply:
I see. I’d expect this logic to then not be conditional, but to take the greater of either the safe area insets or the positioning margins. As is, this code looks like it’s introducing a conditional block of code that could unwittingly conflict with the layout logic you’re referring to.

Reply:
I agree. This is on-par with existing behavior: https://github.com/material-components/material-components-ios/blob/develop/components/Snackbar/src/private/MDCSnackbarOverlayView.m#L276-L280
I tried not to scope creep and start refactoring the display logic too much, which is why I added the conditional.

Reply:
Please either do the refactoring now or immediately after this change then. This is looking like an increasingly brittle set of code.

Reply:
I have put a few hours into trying to simplify the constraints in Snackbar. However, because the requirements for regular size class W+H (iPads) is vastly different than iPhone devices, it is hard to have them share constraints without a conditional. Reason being is that iPhone constrains rely on the left and right margins, whereas iPad devices have a set min/max width. If we want to make this more flexible, we need to chime in design to provide better width logic for iPad devices (rather than hardcoded min=288 max=568). I have opened b/151032266 for tracking